Applications/FileIO/CsvInterface.h

```diff
@@ -36,9 +39,36 @@ namespace FileIO {
 /**
  * Interface for reading CSV file formats.
  */
-class CsvInterface {
+class CsvInterface  : public BaseLib::IO::Writer
+{
 
 public:
+    /// Contructor (only needed for writing files)
+    CsvInterface() {};
+
+    /// Adds an index vector of size s to the CSV file
+    void addIndexVectorForWriting(std::size_t s);
+
+    /// Adds a data vector to the CSV file. All data vectors have to have the same size.
+    /// Vectors will be written in the same sequence they have been added to the interfaceW.
```

Comment:
The sizes are not checked here, but in the write function, which might be too late already (e.g. to be able to trace where the data comes from). Better to have the "equal size" invariant in the class.

Typo? interfaceW. ?